### PR TITLE
feat: support for `Your Library X`

### PR DESCRIPTION
**Pull request by kyrie25**
2nd attempt at this, before Spotify breaks it like #2025 

image
image

### CODE REVIEW

**Comment by kyrie25** [comment]
Seems that people are starting to get Library X as A/B testing so I'll open this as ready for merging.

File: jsHelper/spicetifyWrapper.js [hunk 1]
```diff
@@ -980,101 +980,141 @@ Spicetify.ContextMenu = (function () {
...
-                ),
-            )
-        )
-        reactObjs.push(obj);
-    }
-    return reactObjs;
-}
+Spicetify._cloneSidebarItem = function (list, sidebarIsCollapsed) {
+	function findChild(parent, key, value) {
+		if (!parent.props) return null;
```
Comment (rxri) [inline jsHelper/spicetifyWrapper.js:985]: If you use brackets below for one line ifs, do it there too

**Review by rxri — CHANGES_REQUESTED** [review]

File: jsHelper/spicetifyWrapper.js (hunk 1 above)
Comment (kyrie25) [reply jsHelper/spicetifyWrapper.js:985]: This was unchanged from the original diff so I didn't make any changes.
If you want to refactor it also then I'll do it later.

Comment (rxri) [reply jsHelper/spicetifyWrapper.js]: Well, I prefer no brackets for one line ifs, but it would mean refactoring all javascript files

**Comment by kyrie25** [comment]
More stuff
Spotify_qWIVnUF9EL

**Review by rxri — APPROVED** [review]

**Review by afonsojramos — APPROVED** [review]

**Pull request merged**